lib/train-habitat/connection.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+require 'net/http'
+require 'json'
+require 'train-habitat/platform'
+
+module TrainPlugins
+  module Habitat
+    class Connection < Train::Plugins::Transport::BaseConnection
+      include TrainPlugins::Habitat::Platform
+
+      def initialize(options)
+        if options.dig(:host).nil?
```

Comment:
Huh. Did Rubocop tell you to use `dig`?

Reply:
Forgot to mentioned what I learned yesterday:

The `&.` is a safe navigation operator and doesn't work on Arrays or Hashes. The `#dig` method is only on `Hash#dig` and `Array#dig`.

```
[1] pry(main)> Account = Class.new do
[1] pry(main)*   def name
[1] pry(main)*     'Han Solo'
[1] pry(main)*   end
[1] pry(main)*   def ship
[1] pry(main)*     nil
[1] pry(main)*   end
[1] pry(main)* end
=> Account
[2] pry(main)> account = Account.new
=> #<Account:0x00007fc7c700afa0>
[3] pry(main)> account&.ship
=> nil
[4] pry(main)> account&.name
=> "Han Solo"
[5] pry(main)> options = { ship: 'rogue-one.space' }
=> {:ship=>"rogue-one.space"}
[6] pry(main)> options.dig(:ship)
=> "rogue-one.space"
[7] pry(main)> options&.(:ship)
NoMethodError: undefined method `call' for {:ship=>"rogue-one.space"}:Hash
from (pry):14:in `__pry__'
[8] pry(main)>
```

Reply:
Safe navigation works for arrays and hashes, but you have to change the syntax and it slipped my mind. `{foo: :bar}&.[](:foo)`

Anyway, the point of adding safe nav there is because Ruby will barf if you pass something that doesn't respond to `[]`, like `nil`. So using `dig` here doesn't get us anything. Maybe just `if options.nil? || options[:host].nil?` there. Probably should have a default for the method argument too. `def initialize(options = {})`.

Reply:
@TrevorBramble updated